### DISCUSSION TITLE
Unique constraint on multiple fields

### DISCUSSION THREAD

**Discussion by yisheng** [directus/directus#16909] [Archive]
Hi all,

## Scenario

Suppose we have an `Article` collection, and users (`directus_users` collection) could `like` articles, so here is a M2M relationship. But a user can `like` an article for only one time, so I want to add a unique constraint on two fields.

## Qestion

1. Could we set a unique constraint on multiple columns?
2. What about directly executing `ADD UNIQUE KEY (field1, field2)` sql on mysql?
3. Otherwise, we have to write codes or create a Flow to make this happen.

Thanks.

**Comment by paescuj** [comment]
Duplicate of #3958, #7721, #11424, #13673

**Reply by yisheng** [reply]
Answer: https://github.com/directus/directus/discussions/3958
